earth_enterprise/src/fusion/autoingest/AssetSerializer_unittest.cpp

```diff
@@ -83,14 +116,155 @@ class AssetSerializerTest : public testing::Test {
     // Reset the static variables in TestItem
     TestItem::nextValue = 1;
     TestItem::loadFile = "";
+    TestItem::xmlFilename = "filename.xml";
+    TestItem::newInvalidCalled = false;
     exceptionToThrow = NONE;
+    getFileInfoReturnValue = true;
+    getFileInfoSize = 1000;
+    getFileInfoTime = time(0);
+    geDocPtr = nullptr;
   }
 };
 
-TEST_F(AssetSerializerTest, Load) {
-  auto item = serializer.Load("myfile");
-  ASSERT_EQ(TestItem::loadFile, "myfile");
-  ASSERT_EQ(item->val, 1);
+std::string RunLoadForException(AssetSerializerLocalXML<TestItem> &serializer){
+  string errorMsg = "";
+  try {
+    string boundref = "doesnt_matter";
+    serializer.Load(boundref);
+  }
+  catch (khException e) {
+    errorMsg = string(e.what());
+  }
+
+  return errorMsg;
+}
+
+TEST_F(AssetSerializerTest, Load_GetFileInfoFails) {
+  getFileInfoReturnValue = false;
+  string errorMsg = RunLoadForException(serializer);
+  ASSERT_EQ(errorMsg, "No such file: filename.xml");
+}
+
+TEST_F(AssetSerializerTest, Load_FileSizeZero) {
+  getFileInfoSize = 0;
+  string errorMsg = RunLoadForException(serializer);
+  ASSERT_EQ(errorMsg, "No such file: filename.xml");
+}
+
+TEST_F(AssetSerializerTest, Load_InvalidXMLDoc) {
+  string errorMsg = RunLoadForException(serializer);
+  ASSERT_EQ(errorMsg, "Unable to read filename.xml");
+}
+
+TEST_F(AssetSerializerTest, Load_NoDocElement) {
+  geDocPtr = std::unique_ptr<GEDocument>(new GECreatedDocument(""));
+  ASSERT_FALSE(geDocPtr == nullptr);
+  string errorMsg = RunLoadForException(serializer);
+  ASSERT_EQ(errorMsg, "Error loading filename.xml: No document element");
+}
+
+TEST_F(AssetSerializerTest, Load_UnknownAssetType) {
+  string docElement = "DummyAsset";
+  geDocPtr = CreateEmptyDocument(docElement);
+  ASSERT_FALSE(geDocPtr == nullptr);
```

Comment:
Minor: there's an `ASSERT_NE` for not equal tests.

Reply:
I think that was brain fart, because I'm using  `ASSERT_NE` other places. Also, I meant to take this assert out anyway. I was using that to make sure that `CreateEmptyDocument` worked like I expected. 
I'll remove this and the other, similar one.